### PR TITLE
add lesson19

### PR DESCRIPTION
**Pull request by haraldF**

### CODE REVIEW

**Review by carlotrimarchi — APPROVED** [review]

File: lesson19.md [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+```
+
+---
+
+### Null
+
+* There's one datatype that we haven't learned about yet: `null`
+* `null` is like `undefined`. It indicates the absence of a value.
+* `null` can be used to indicate an *intentional* absence of an object value.
+* Example: `document.getElementById("NoExist");` returns `null`
```
Comment (pektinasen) [inline lesson19.md:42]: insert tony-hoare-crying-meme here :D

File: lesson19.md [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+   * `null`
+   * another JSON `object`
+* These types are **forbidden**:
+   * `function`, `undefined`
+
+---
+
+### Why JSON?
+
+* When we exchange data between a browser and a server, we can only use strings (not objects, numbers, booleans...)
```
Comment (pektinasen) [inline lesson19.md]: i'm pretty sure i can exchange binary data, as well

File: lesson19.md [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+![API](images/API.png) <!-- .element height="400px" width="400px" -->
+
+---
+
+### REST API
+
+* REST stands for **Re**presentational **S**tate **T**ransfer
+* REST APIs work via http(s), the same protocol we use to connect to any website
+* REST APIs commonly use JSON to exchange data
+* Example for a REST API call: https://restcountries.eu/rest/v2/lang/de
```
Comment (pektinasen) [inline lesson19.md]: could be unclear what you mean, because this just looks like a url

File: lesson19.md [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+### Practice
+
+* Use the free API from https://restcountries.eu
+* Example API call:
+   * https://restcountries.eu/rest/v2/name/germany
+* Look at the result
+* Use the `fetch` API to fetch the JSON.
+* Can you show the name, capital, population and the flag on your web page?
+* BONUS: Add an `<input>` field where the user can enter a country name, and a "search" button. When the user clicks "search", `fetch` ahd show the details about the country in the input field.
```
Comment (sevtapim) [inline lesson19.md]:
```suggestion
* BONUS: Add an `<input>` field where the user can enter a country name, and a "search" button. When the user clicks "search", `fetch` and show the details about the country in the input field.
```

Comment (sevtapim) [inline lesson19.md]:
```suggestion
* BONUS: Add an `<input>` field where the user can enter a country name, and a "search" button. When the user clicks "search", `fetch` and show the details about the country in the input field.
```

**Review by sevtapim — COMMENTED** [review]
spelling error

File: lesson19.md (hunk 2 above)
Comment (haraldF) [reply lesson19.md]: strings can contain binary data. At least that's what we had to use before those fancy arrayBuffers :)

Comment (pektinasen) [reply lesson19.md]: mmh. can't I send a file via js as binary data? or do I have to decode it as base64 or something?

But then yes, nowadays there is probably some implementation to send protobuf messages from the browser, right?

Comment (haraldF) [reply lesson19.md]: It's complicated :)

https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Sending_and_Receiving_Binary_Data#Receiving_binary_data_in_older_browsers

Comment (pektinasen) [reply lesson19.md]: :sob:

Comment (haraldF) [reply lesson19.md]: I've reworded it a bit, hope it's less ambiguous :)

File: lesson19.md (hunk 3 above)
Comment (haraldF) [reply lesson19.md]: Reworded

**Pull request merged**